pkg/ddl/sanity_check.go

```diff
@@ -46,7 +46,14 @@ func (e *executor) checkDeleteRangeCnt(job *model.Job) {
 		panic(err)
 	}
 	if actualCnt != expectedCnt {
-		panic(fmt.Sprintf("expect delete range count %d, actual count %d", expectedCnt, actualCnt))
+		switch job.Type {
+		case model.ActionReorganizePartition, model.ActionRemovePartitioning, model.ActionAlterTablePartitioning:
+			if actualCnt < expectedCnt {
```

Comment:
We could, but since I added tests to verify that no old partitions or indexes contain data after GC, I don't think it is needed to replicate the full logic of cleaning up global indexes etc. Checking that non of the previous indexes and partitions exists in KV store after the DDL should be OK.

Reply:
I think modify `expectedDeleteRangeCnt` is better. This looks more integrated.

Reply:
> I think modify expectedDeleteRangeCnt is better.

Agree. This is more like a test, it is better to change the test itself instead of the assertion.

Reply:
OK, I will update the logic in `expectedDeleteRangeCnt()`.